tests/test_schemathesis.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+import schemathesis
+from fastapi import FastAPI
+from schemathesis.specs.openapi.schemas import BaseOpenAPISchema
+
+
+@pytest.fixture()
+def app_schema(client) -> BaseOpenAPISchema:
+    """
+    Get an OpenAPI schema instance for the app created by the `client` fixture.
+    """
+    app: FastAPI = client.app
+    assert client.app is not None
+    openapi = app.openapi()
+    result = schemathesis.from_dict(openapi)
+    return result
+
+
+schema = schemathesis.from_pytest_fixture("app_schema")
+
+
+@schema.parametrize()
+def test_api(case):
+    """Run tests automatically generated by schemathesis."""
+    case.call_and_validate()
```

Comment:
The default `call_and_validate` uses `requests.Session` that is not aware of the ASGI app.

Having the latest version of the project locally with the following changes to this test runs Schemathesis as expected:

```python
@schema.parametrize()
def test_api(case, client):
    """Run tests automatically generated by schemathesis."""
    case.call_and_validate(session=client)
```

Though, the tests are failing as it finds some inconsistencies.

Reply:
I've incorporated this change. Thanks @Stranger6667.